llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -661,7 +661,8 @@ bool VPInstruction::isFPMathOp() const {
   return Opcode == Instruction::FAdd || Opcode == Instruction::FMul ||
          Opcode == Instruction::FNeg || Opcode == Instruction::FSub ||
          Opcode == Instruction::FDiv || Opcode == Instruction::FRem ||
-         Opcode == Instruction::FCmp || Opcode == Instruction::Select;
+         Opcode == Instruction::FCmp || Opcode == Instruction::Select ||
+         Opcode == VPInstruction::WideIVStep;
```

Comment:
Will this return true even when the type operand is an integer?

Reply:
Oh good point. I had run into this assertion with a wide step with an integer type in `VPInstruction::execute`:

```c++
  assert((hasFastMathFlags() == isFPMathOp() ||
          getOpcode() == Instruction::Select) &&
         "Recipe not a FPMathOp but has fast-math flags?");
```

But this opcode shouldn't have been executed in the first place anyway, it was due to an issue on my end!